pkg/nfd-master/nfd-master.go

```diff
@@ -290,13 +290,14 @@ func updateNodeFeatures(helper apihelper.APIHelpers, nodeName string, labels Lab
 	removeLabelsWithPrefix(node, "node.alpha.kubernetes-incubator.io/node-feature-discovery")
 
 	// Add labels to the node object.
+	filterFeatureLabels(&labels, s.args.LabelWhiteList)
```

Comment:
I think we must filter the labels before the annotation is generated no? Here we are adding all the labels in the annotation and then filter the labels using the regex.

Reply:
Yup, you're correct 🤭 I obviously wrote this patch a bit too hastily.

This is now fixed and the unit test is updated to check this. I also added another patch fixing a former issue in unit tests